lib/cocoapods/executable.rb

```diff
@@ -77,6 +75,27 @@ def self.execute_command(executable, command, raise_on_failure)
       output
     end
 
+    private
+
+    def self.popen3(bin, command, stdout, stderr)
+      require 'open3'
+      Open3.popen3(bin, *command) do |i, o, e, t|
+        out_reader = Thread.new { while s = o.gets; stdout << s; end }
+        err_reader = Thread.new { while s = e.gets; stderr << s; end }
+
+        i.close
+
+        run_readers = lambda do
+          [out_reader, err_reader].each do |reader|
+            reader.run if reader.alive?
+          end
+        end
+
+        run_readers.call
+        t.value.tap { run_readers.call }
```

Comment:
no, just using `tap` here since we want to return `t.value`, but we want to let the readers have one last go at the buffers if it turns out the wait thread finished before the readers